runtime/src/config.c

```diff
@@ -431,48 +432,54 @@ int handlePossibleConfigVar(int* argc, char* argv[], int argnum,
     }
   }
 
-  chpl_mem_free(argCopy, argnum, "<command-line>");
+  chpl_mem_free(argCopy, argnum, CHPL_FILE_IDX_COMMAND_LINE);
   return retval;
 }
 
 // TODO: Change all the 0 linenos below into real line numbers
 void parseConfigFile(const char* configFilename, 
-                     int32_t lineno, c_string filename) {
+                     int32_t lineno, int32_t filename) {
   FILE* argFile = fopen(configFilename, "r");
   if (!argFile) {
     char* message = chpl_glom_strings(2, "Unable to open ", configFilename);
     chpl_error(message, lineno, filename);
   }
+  chpl_saveFilename(configFilename); // CHPL_FILE_IDX_SAVED_FILENAME will now
+                                     // give us configFilename
   while (!feof(argFile)) {
     int numScans = 0;
     char setConfigBuffer[_default_string_length];
     numScans = fscanf(argFile, _default_format_read_string, setConfigBuffer);
     if (numScans == 1) {
-      if (!aParsedString(argFile, setConfigBuffer, 0, configFilename)) {
+      if (!aParsedString(argFile, setConfigBuffer, 0,
+                         CHPL_FILE_IDX_SAVED_FILENAME)) {
         char* equalsSign;
         const char* moduleName;
         char* varName;
-        configVarType* configVar = breakIntoPiecesAndLookup(setConfigBuffer, 
-                                                            &equalsSign,
-                                                            &moduleName, 
-                                                            &varName, 
-                                                            0, configFilename);
+        configVarType *configVar =
+            breakIntoPiecesAndLookup(setConfigBuffer, &equalsSign, &moduleName,
```

Comment:
Runtime indentation is 2 spaces per level rather than 4. There are some other instances of this elsewhere in this file. I don't recall seeing it in the other runtime changes you've made.